src/app/modules/feedback/feedback.controller.ts

```diff
@@ -0,0 +1,143 @@
+import { celebrate, Joi, Segments } from 'celebrate'
+import { StatusCodes } from 'http-status-codes'
+import { ErrorDto, PrivateFormErrorDto } from 'shared/types'
+
+import { createLoggerWithLabel } from '../../config/logger'
+import { createReqMeta } from '../../utils/request'
+import { ControllerHandler } from '../core/core.types'
+import { PrivateFormError } from '../form/form.errors'
+import * as FormService from '../form/form.service'
+import * as PublicFormService from '../form/public-form/public-form.service'
+import { mapRouteError } from '../form/public-form/public-form.utils'
+import * as SubmissionService from '../submission/submission.service'
+
+import * as FeedbackService from './feedback.service'
+
+const logger = createLoggerWithLabel(module)
+
+const validateSubmitFormFeedbackParams = celebrate({
+  [Segments.BODY]: Joi.object()
+    .keys({
+      rating: Joi.number().min(1).max(5).cast('string').required(),
+      comment: Joi.string().allow('').required(),
+    })
+    // Allow other keys for backwards compability as frontend might put extra keys in the body.
+    .unknown(true),
+})
+
+export const submitFormFeedback: ControllerHandler<
+  { formId: string; submissionId: string },
+  { message: string } | ErrorDto | PrivateFormErrorDto,
+  { rating: number; comment: string }
+> = async (req, res) => {
+  const { formId, submissionId } = req.params
+  const { rating, comment } = req.body
+  const logMeta = {
+    action: 'submitFormFeedback',
+    ...createReqMeta(req),
+    formId,
+    submissionId,
+  }
+
+  const checkDoesSubmissionIdExistRes =
+    await SubmissionService.checkDoesSubmissionIdExist(submissionId)
+  if (checkDoesSubmissionIdExistRes.isErr()) {
+    const { error } = checkDoesSubmissionIdExistRes
+    logger.error({
+      message: 'Failed to check if submissionId exists',
+      meta: logMeta,
+      error,
+    })
+    const { errorMessage, statusCode } = mapRouteError(error)
+    return res.status(statusCode).json({ message: errorMessage })
+  }
+
+  const isSubmissionIdValid = checkDoesSubmissionIdExistRes.value
+  if (!isSubmissionIdValid) {
+    return res
+      .status(StatusCodes.NOT_FOUND)
+      .json({ message: 'SubmissionId is not valid' })
+  }
+
+  const checkHasPreviousFeedbackRes =
+    await FeedbackService.checkHasPreviousFeedback(formId, submissionId)
+  if (checkHasPreviousFeedbackRes.isErr()) {
+    const { error } = checkHasPreviousFeedbackRes
+    logger.error({
+      message:
+        'Failed to check if feedback has already been submitted previously',
+      meta: logMeta,
+      error,
+    })
+    const { errorMessage, statusCode } = mapRouteError(error)
+    return res.status(statusCode).json({ message: errorMessage })
+  }
+
+  const hasPreviousFeedback = checkHasPreviousFeedbackRes.value
+  if (hasPreviousFeedback) {
+    return res
+      .status(StatusCodes.UNPROCESSABLE_ENTITY)
+      .json({ message: 'Multiple feedbacks has already been submitted' })
```

Comment:
similarly not a helpful error message for the user to see. maybe something like 'You have already submitted feedback for this submission. To provide additional feedback, please contact the form administrator."